evalml/tests/model_understanding_tests/test_permutation_importance.py

```diff
@@ -0,0 +1,274 @@
+                                'OHE_1': {'features_to_encode': ['currency', 'expiration_date', 'provider']},
+                                'OHE_2': {'features_to_encode': ['region', 'country']}}),
+              (DagReuseFeatures, {'SelectNumeric': {'columns': ['card_id', 'store_id', 'datetime', 'amount', 'customer_present', 'lat', 'lng']},
+                                  'SelectCategorical1': {'columns': ['currency', 'expiration_date', 'provider']},
+                                  'SelectCategorical2': {'columns': ['region', 'country']},
+                                  'OHE_1': {'features_to_encode': ['currency', 'expiration_date', 'provider']},
+                                  'OHE_2': {'features_to_encode': ['region', 'country']},
+                                  'DT': {'encode_as_categories': True}}),
+              (LinearPipelineWithTargetEncoderAndOHE, {'One Hot Encoder': {'features_to_encode': ['currency', 'expiration_date', 'provider']},
+                                                       'Target Encoder': {'cols': ['region', 'country']}}),
+              (LinearPipelineCreateFeatureThenDropIt, {'Select Columns Transformer': {'columns': ['amount']},
+                                                       'DoubleColumns': {'drop_old_columns': False},
+                                                       'Drop Columns Transformer': {'columns': ['amount_doubled']}})
+              ]
+
+
+@pytest.mark.parametrize('pipeline_class, parameters', test_cases)
+@patch('evalml.pipelines.PipelineBase._supports_fast_permutation_importance', new_callable=PropertyMock)
+def test_fast_permutation_importance_matches_sklearn_output(mock_supports_fast_importance, pipeline_class, parameters,
+                                                            has_minimal_dependencies):
+    if has_minimal_dependencies and pipeline_class == LinearPipelineWithTargetEncoderAndOHE:
+        pytest.skip("Skipping test_fast_permutation_importance_matches_sklearn_output for target encoder cause "
+                    "dependency not installed.")
+    X, y = load_fraud(100)
+
+    # Do this to make sure we use the same int as sklearn under the hood
+    random_state = np.random.RandomState(0)
+    random_seed = random_state.randint(np.iinfo(np.int32).max + 1)
+
+    mock_supports_fast_importance.return_value = True
+    parameters['Random Forest Classifier'] = {'n_jobs': 1}
+    pipeline = pipeline_class(parameters=parameters)
+    pipeline.fit(X, y)
+    fast_scores = calculate_permutation_importance(pipeline, X, y, objective='Log Loss Binary',
+                                                   random_state=random_seed)
+
+    mock_supports_fast_importance.return_value = False
+    slow_scores = calculate_permutation_importance(pipeline, X, y, objective='Log Loss Binary',
+                                                   random_state=0)
+
+    pd.testing.assert_frame_equal(fast_scores, slow_scores)
```

Comment:
Test looks good! Since we're ideally getting a speed boost by doing faster performance, would we be able to do assertions that the time for `fast_scores` is <= time for `slow_scores`?

Reply:
Good point! I tried a few times locally and the time for fast scores is smaller than slow scores. I'm pushing this up and keeping an eye if the results vary from run to run.

Reply:
It flaked on windows so I'm reverting it lol.

Reply:
classic windows lol